File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/iterator/BlogPlatform.java
package iterator;

import java.util.ArrayList;
import java.util.List;

class BlogPlatform implements Blog {
    private List<Post> posts;

    public BlogPlatform() {
        this.posts = new ArrayList<>();
    }

    public void addPost(Post post) {
        posts.add(post);
    }

    @Override
    public Iterator createIterator() {
        return new PostIterator(posts);
    }
}

File: code/csharp/DesignPatterns/iterator/BlogPlatform.cs
using System.Collections.Generic;

namespace Iterator
{
    public class BlogPlatform : IBlog
    {
        private List<Post> _posts = new List<Post>();

        public void AddPost(Post post)
        {
            _posts.Add(post);
        }

        public IIterator CreateIterator()
        {
            return new PostIterator(_posts);
        }
    }
}
